etc/pr_check/fake_github.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+type fakeGitHub struct {
+	*http.ServeMux
+	prs          []map[string]interface{}
+	pagesFetched int
+}
+
+func newFakeGitHub() *fakeGitHub {
+	r := &fakeGitHub{
+		ServeMux: http.NewServeMux(),
+		prs:      make([]map[string]interface{}, 0, 25),
+	}
+	fakePRs, err := os.Open("fake_github_prs.json")
+	if err != nil {
+		panic("could not open fake_github_prs.json:\n" + err.Error())
+	}
+	json.NewDecoder(fakePRs).Decode(&r.prs)
+
+	r.HandleFunc("/repos/pachyderm/pachyderm/pulls", r.listHandler)
+	r.HandleFunc("/", r.defaultHandler)
+	return r
+}
+
+func (f *fakeGitHub) listHandler(w http.ResponseWriter, r *http.Request) {
+	// Get params from request (mostly 'page', but check 'per_page')
+	var (
+		direction      = "desc"
+		page           = 1
+		resultsPerPage = 5 // real GitHub is 30 by default
+		err            error
+	)
+	if r.URL.Query().Has("direction") {
+		direction = r.URL.Query().Get("direction")
+	}
+	getIntParam := func(key string, defaultVal int) (int, error) {
+		if r.URL.Query().Has(key) {
+			result, err := strconv.Atoi(r.URL.Query().Get(key))
+			if err != nil {
+				return 0, fmt.Errorf("invalid parameter value for '%s': could not be parsed as an int (%q)", key, err)
+			}
+			return result, nil
+		}
+		return defaultVal, nil
+	}
+	if page, err = getIntParam("page", page); err != nil {
```

Comment:
I prefer the 2-line approach where an existing variable is being overwritten:

```go
page, err = getIntParam("page", page)
if err != nil {
...
}
```

I think that `if x, err = ...; err != nil {` and `if x, err := ...; err != nil` are too visually similar with dramatically different side effects.  So, I only use `if` with `:=`.  Using the wrong form has caused at least one security error in the auth code (for Hub).